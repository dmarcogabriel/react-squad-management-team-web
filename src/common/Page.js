import React from 'react';
import Header from './Header';
import Footer from './Footer';

const Page = ({ children }) => (
  <>
    <Header />

    <div className="flex flex-col md:flex-row bg-background justify-center w-full px-5 md:px-10">
      {children}
    </div>

    <Footer />
  </>
);

export default Page;
