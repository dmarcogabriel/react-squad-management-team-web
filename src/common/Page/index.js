import React from 'react';
import cn from 'classnames';
import Header from '../Header';
import Footer from '../Footer';

const Page = ({ children }) => (
  <>
    <Header />

    <div
      data-testid="page"
      className={cn(
        'flex flex-col md:flex-row justify-center',
        'bg-background w-full px-5 md:px-10'
      )}
    >
      {children}
    </div>

    <Footer />
  </>
);

export default Page;
