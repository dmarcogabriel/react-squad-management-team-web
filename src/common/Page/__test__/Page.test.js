import { render } from '@testing-library/react';
import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Page from '..';

describe('<Page />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Page />, div);
  });

  it('renders classes correctly', () => {
    const { getByTestId } = render(<Page />);

    expect(getByTestId('page')).toHaveClass(
      'flex flex-col md:flex-row bg-background justify-center w-full px-5 md:px-10'
    );
  });

  it('matches snapshot', () => {
    expect(
      shallow(
        <Page>
          <div>
            <h1>This is a header 1 from Page</h1>
          </div>
        </Page>
      )
    ).toMatchSnapshot();
  });
});
