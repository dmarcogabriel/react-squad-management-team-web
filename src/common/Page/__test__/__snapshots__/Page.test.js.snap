// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> matches snapshot 1`] = `
Array [
  <nav
    className="flex items-center justify-between px-8 py-2 text-white bg-gradient-to-r from-primary-dark to-secondary-dark"
    data-testid="header"
  >
    <div>
      <p
        className="text-lg"
      >
        Squad Management Tool
      </p>
    </div>
    <div
      className="flex items-center"
    >
      <p
        className="text-sm mr-2 invisible md:visible"
      >
        John Doe
      </p>
      <p
        className="bg-white py-2 px-3 rounded-full text-black"
      >
        JD
      </p>
    </div>
  </nav>,
  <div
    className="flex flex-col md:flex-row justify-center bg-background w-full px-5 md:px-10"
    data-testid="page"
  >
    <div>
      <h1>
        This is a header 1 from Page
      </h1>
    </div>
  </div>,
  <div
    className="bg-footer py-3 flex justify-center"
    data-testid="footer"
  >
    <p
      className="text-sm text-gray-900"
      data-testid="footer-text"
    >
      2020 - All rights reserved
    </p>
  </div>,
]
`;
